File: Benchmarking/Euler_116/euler116.py
import sys

def solve(m, n): #m=color block size  n = black box size
  if m > n:
    return 0
  solutions = 0

  for i in xrange(m, n+1):
    solutions += 1
    solutions += solve(m, n-i)

  return solutions

def s(size):
  result = 0;
  for i in xrange(0, 3):
     result += solve(2+i, size)
  return result

size = int(sys.argv[1])

s(size)

File: report/poster/euler116.jl
function solve(tileSize, blockSize) #m=color block size  n = black box size
  if tileSize > blockSize
    return 0
  end
  solutions = 0

  for i = tileSize : blockSize
    solutions += 1
    solutions += solve(tileSize, blockSize-i)
  end

  return solutions
end

function calc(size)
  result = solve(2, size) #Red tiles
  result += solve(3, size) #Green tiles
  result += solve(4, size) #Blue tiles

  return result
end

size = parse(Int32, ARGS[1])

calc(size)